### PR TITLE
Remove json config code

### LINKED ISSUE #3530: Remove unnecessary json serializing of config files
With previous `.json` config files being replaced in V20.0 with `.toml` files, we kept the related `nano::logging::serialize_json` code to support upgrade paths from V19.0 forward. But now that V19.0 is an [inactive release](https://docs.nano.org/releases/node-releases/#inactive-releases) we should verify this serialization code and other code related to processing `.json` config files is removed and cleaned up.

### PR DESCRIPTION
**Pull request by cryptocode**
This improves compile/test times a bit and removes unnecessary code. 

The json->toml upgrade paths are removed as well. In the unlikely case someone needs to upgrade from v19.x and earlier, the solution is to run a v2x node (prior to this change) to perform the upgrade.

Closes #3530

### CODE REVIEW

**Review by clemahieu — DISMISSED** [review]

File: nano/core_test/work_pool.cpp [hunk 1]
```diff
@@ -115,21 +114,6 @@ TEST (work, opencl)
 	}
 }
 
-TEST (work, opencl_config)
```
Comment (dsiganos) [inline nano/core_test/work_pool.cpp:118]: Should this test case be removed?
Can we convert it to use toml and keep it?

File: nano/rpc_test/rpc.cpp [hunk 2]
```diff
@@ -5592,40 +5592,6 @@ TEST (rpc, in_process)
 	ASSERT_EQ ("0", pending_text);
 }
 
-TEST (rpc_config, serialization)
```
Comment (dsiganos) [inline nano/rpc_test/rpc.cpp:5595]: I think this test case should be converted to toml and kept.
I do not see an equivalent one for toml.

**Comment by dsiganos** [comment]
Should we remove `nano::stat_config::deserialize_json` also?

**Review by theohax — DISMISSED** [review]

File: nano/core_test/work_pool.cpp (hunk 1 above)
Comment (cryptocode) [reply nano/core_test/work_pool.cpp:118]: Yes should be removed as this is covered by the TOML deserialization tests which have full coverage of all properties, including opencl. The serialization part is irrelevant for TOML.

File: nano/rpc_test/rpc.cpp (hunk 2 above)
Comment (cryptocode) [reply nano/rpc_test/rpc.cpp:5595]: There's never any serialization to TOML (except when generating defaults - unless this has been changed recently).Deserialization is covered by `toml, rpc_config_deserialize_no_defaults` and `rpc_config_no_required`

**Comment by cryptocode** [comment]
> Should we remove `nano::stat_config::deserialize_json` also?

Yeah good point, will push an update.

File: nano/lib/stats.cpp [hunk 3]
```diff
@@ -10,36 +10,6 @@
 #include <fstream>
 #include <sstream>
 
-nano::error nano::stat_config::deserialize_json (nano::jsonconfig & json)
```
Comment (dsiganos) [inline nano/lib/stats.cpp:13]: The header file declaration needs removing too.
`nano::error deserialize_json (nano::jsonconfig & json);`

**Comment by dsiganos** [comment]
And I think we can also remove jsonconfig::read_and_update().

**Review by dsiganos — APPROVED** [review]

**Comment by dsiganos** [comment]
Looks good to me! Thank you for doing the amendments!

**Review by clemahieu — APPROVED** [review]
Very nice cleanup. Thank you for pushing this through!

**Pull request merged**